sqlmesh/core/table_diff.py

```diff
@@ -277,6 +277,20 @@ def name(e: exp.Expression) -> str:
                         1,
                         0,
                     ).as_("row_joined"),
+                    exp.func(
+                        "IF",
+                        exp.or_(
+                            *(
+                                exp.and_(
```

Comment:
will s and t ever both be NULL? is this correct? have you tested it

Reply:
Yes I tested it and since this query is performed on the resulting table of the full join, a null primary key in table 's' will have a null values for the primary key as well in table 't' in the same row:

```
item_id │ num_orders │ item_id │ num_orders │
────────│────────────│─────────│────────────
   int32 │     int64  │   int32 │     int64
────────│────────────│─────────│────────────
      1 │         5  │       1 │         5
      2 │         1  │       2 │         1
        │            │         │         3
        │            │         │         2
```

So I couldn't think of another way to construct this case to include it in the main query for the fully joined table. Except if the check for null values was with a separate query on the individual tables themselves

Reply:
hey @Themiscodes can you explain this condition to me? you're trying to check that any of the index cols are null right. why are you using an and condition?

should it be if( s_a is null or s_b is null or t_a is null or t_b is null) ?

Reply:
The issue with the condition `s_a is null or s_b is null or t_a is null or t_b` is it would evaluate to `True` in cases where there are no actual nulls in the original tables, since the query is applied to the `fully joined` table which fills nulls in the rows that are not matching.

If for example we had the tables:

**Prod:**
```
┌─────────┬────────────┐
│ item_id │ num_orders │
│  int32  │   int64    │
├─────────┼────────────┤
│       1 │          5 │
│       3 │          1 │
│       2 │          1 │
└─────────┴────────────┘
```

**Dev:**
```
┌─────────┬────────────┐
│ item_id │ num_orders │
├─────────┼────────────┤
│       3 │          1 │
│       1 │          5 │
│       2 │          1 │
│       4 │          6 │
└─────────┴────────────┘
```

The resulting full join would be:
```
┌───────────┬──────────────┬───────────┬──────────────┐
│ s__item_id│ s__num_orders│ t__item_id│ t__num_orders│
├───────────┼──────────────┼───────────┼──────────────┤
│        1.0│           5.0 │         1 │            5 │
│        3.0│           1.0 │         3 │            1 │
│        2.0│           1.0 │         2 │            1 │
│        NaN│           NaN │         4 │            6 │
└───────────┴──────────────┴───────────┴──────────────┘
```

So having the `and` in the condition prevents false positives like this. And since if a key is null in one table, it will also be null in the same row of the full join table, actual null keys are properly identified